### DISCUSSION TITLE
How to change/force certain language and landing page

### DISCUSSION THREAD

**Discussion by ekrocks** [bullhorn/career-portal#424] [Q&A] [answered]
I would like to know if there is any documentation on how to set certain filters to be enabled when user clicks on the link to the career portal url. 

For example, if the user is using the website from Central Canada, I would like the filters to be automatically enabled for jobs in Central Canada. 

The second question ties into this as well. Currently the site automatically chooses the locale for the site but I would like to allow users to choose the locale. Is there a way to alter the .json file or the jobOrder query such that it allow users to change the language to their needs?

**Comment by charlesabarnes** [comment, marked as answer]
This would require reconfiguration of the `SettingService` within the app, Currently, the app does not request location from the user, but can be modified to do so.  Logic can be found for the locale choosing logic here https://github.com/bullhorn/career-portal/blob/master/src/app/services/settings/settings.service.ts.   you can probably instead add a dropdown, save their choice to localStorage, then refresh the page and pull the preference from there.

Please let me know if you have any more questions.

**Reply by ekrocks** [reply]
Thank you for the response, just another question regarding the locale settings; is there any settings that creates a different url depending on different locales? for example www.examplewebsite.ca/en-us or www.examplewebsite.ca/fr-FR? Or is all the settings done internally and no url changes are possible?

**Reply by charlesabarnes** [reply]
There aren't any settings to do that.  The only suggestion I can make for that setup, is to create the 2 portals in their respective directories with the settings modified

**Reply by pbearne** [reply]
hi @ekrocks we have had great success doing Multi-language sites with matador jobs as we can leverage the Multi-language plugins that WordPress has

**Reply by ekrocks** [reply]
I just got back with more requests on different landing pages and I realized I created this ticket a while ago. I had opened a ticket with Bullhorn and they lead me back to the GitHub to ask the technical questions here. 
- Currently our site has different pages for people from different locations (in which they can select manually).
- Afterwards, the user can access the portal site from their respective location in which then the portal will view the careers in that State.

I took a look at ```SettingService``` but I think you pointed at that file for the locale settings. 
Is there any way to save the filters to certain selection such that when user will immediately be shown the jobs in their state?

so if someone clicks "New Brunswick" 
![image](https://user-images.githubusercontent.com/10538258/139087791-270448f9-e701-44a4-8a05-9b392311fdce.png)
from our website, it will set the filters like this?
![image](https://user-images.githubusercontent.com/10538258/139087908-13eef117-be5f-4d1c-9b8c-bb9940047862.png)
when the user clicks on a link to go into the Career Portal?
